extensions/api/observability/README.md

```diff
@@ -0,0 +1,35 @@
+# Connector Observability
+
+### `/check/health` endpoint
+
+returns information about the system status. This is primarily intended to be used
+in [`HEALTHCHECK` instructions in Docker files](https://docs.docker.com/engine/reference/builder/#healthcheck). Here,
+this endpoint always returns HTTP 200 indicating that the system is healthy as soon as this endpoint is reachable.
+
+### `/check/liveness` endpoint
+
+From the [Kubernetes docs](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#container-probes):
+> indicates whether the container is running.
+
+Basically the same semantics as the `/health` endpoint. We can assume, that in the event that the connector crashes, all
+REST endpoints become unavailable.
+
+### `/check/readiness` endpoint
+
+From the [Kubernetes docs](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#container-probes):
+> Indicates whether the container is ready to respond to requests.
+
+Thus, the readiness endpoint must return a value indicating success _only after_ all extensions that contribute an API
+have started successfully. Note, that other subsystems like the catalog crawlers might not yet be completed at that
+point, but technically the connector is "ready to respond to requests", even if they might not produce the desired - or
+even a sensible - outcome. Catalog queries may produce an empty or incomplete result until crawlers have completed.
+
+### `/check/startup` endpoint
```

Comment:
I don't think a separate endpoint (and separate registration mechanism) is needed for startup. We can just use the standard health endpoint for startup. See example at https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-startup-probes

For example, in our PRS application we use this because of the time it takes for our Spring Boot app to start and Flyway to seed the database. See https://github.com/catenax/tractusx/blob/3565aeee25b34ebc23febb431f6cceddc41955fc/coreservices/partsrelationshipservice/cd/helm/prs/templates/deployment.yaml#L67

Flyway is just another health check in Spring Boot.

Reply:
There may be a subtle difference between the health endpoint and the startup endpoint as they are defined now: while the health endpoint becomes available as soon as the `observability`-extension gets loaded, the system startup may not yet have completed. The startup endpoint should only return `success` once _all_ extensions have completed loading. 

Although not everyone may use the `startup` endpoint, there are certain use cases where it comes in handy, e.g. during testing: the sanity checks could leverage it

Reply:
Yes we agree on this. My point is that, in every application I've seen, the health and the startup probe use the same endpoint. They're simply configured differently in Kubernetes, e.g. to kill the pod if unavailable for 1 minute but give it 5 minutes to be available when it starts up.

So while these may be two different *probes* in the pod configuration, it is actually one single *endpoint* in the application itself, i.e. are you up, ready and capable of serving requests.

Reply:
I was actually going to make `liveness` and `health` the same, but sticking to best practice also makes sense, so I will change it to `startup == health`.